unlock-app/src/components/creator/lock/EmbedCodeSnippet.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import styled from 'styled-components'
+import UnlockPropTypes from '../../../propTypes'
+
+export function EmbedCodeSnippet({ lock }) {
+  function embedCode(lock) {
+    return `<!-- Include this script in your <head> section -->
+<script src="https://unlock-protocol.com/unlock.js"></script>
+
+<!-- Lock elements by wrapping them in this div -->
+<div unlock-lock="${lock.address}">
+  This content is only visible to the visitors who have a key to the lock
+  ${lock.address}
+</div>
+`
+  }
+
+  function selectAll(event) {
+    event.target.select()
+  }
+
+  return (
+    <CodeSnippet value={embedCode(lock)} onClick={selectAll}></CodeSnippet>
+  )
+}
+
+EmbedCodeSnippet.propTypes = {
+  lock: UnlockPropTypes.lock,
+}
+
+export default EmbedCodeSnippet
+
+const CodeSnippet = styled.textarea`
+  width: 100%;
+  height: 163px;
```

Comment:
hum, do you need to block the height?

Reply:
Also is `width: 100%;` required too? I worry this will behave strangely in grid system.

Reply:
you probably want to use the `grid-column-template: 1fr <whatever you need for the button>`

Reply:
So I do have that for the container element - the 100% here just ensures that it fills up the grid column. The height is also required, because otherwise the textarea is roughly two lines tall. The grid adapts to the dimensions.

Reply:
I will let you merge this but I want to insist on the fact that there must be another way to do this. % ages and grids don't work well together and end up causing sizing issues.

Reply:
I refactored this to sit in the grid description, rather than explicitly defining the textarea element size.